addons/purchase_repair/models/repair_order.py

```diff
@@ -0,0 +1,48 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import models, fields, _
+
+
+class RepairOrder(models.Model):
+    _inherit = 'repair.order'
+
+    purchases_count = fields.Integer(compute="_compute_mto_purchases", groups="purchase.group_purchase_user")
+    purchase_mto_ids = fields.Many2many(
+        'purchase.order', string='Purchases', groups='mrp.group_purchase_user', compute='_compute_mto_purchases',
+    )
+
+    def _compute_mto_purchases(self):
+        self.purchase_mto_ids = self._get_mto_links()
+        self.purchases_count = len(self.purchase_mto_ids)
+
+    def _get_mto_links(self):
+        product_ids = [p.id for p in self.move_ids.filtered(lambda x: x.repair_line_type == 'add').product_id]
+        domain = [
+            ('state', 'not in', ['cancel']),
+            ('product_id', 'in', product_ids),
+        ]
+        if self.location_id.warehouse_id:
+            domain += [(
+                'order_id.picking_type_id.warehouse_id',
+                '=', self.location_id.warehouse_id.id
+            )]
+        po_lines = self.env['purchase.order.line'].search(domain)
+        return po_lines.mapped("order_id")
```

Comment:
Is there no stronger procurement link somewhere? i.e. something like move_dest_ids = the repair lines?

This logic will show POs that weren't created via a MTO from the specific repair order (unless that's what mgm wants)